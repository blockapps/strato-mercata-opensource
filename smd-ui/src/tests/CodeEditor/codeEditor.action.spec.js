import {
  compileCodeFromEditor,
  compileCodeFromEditorFailure,
  compileCodeFromEditorSuccess,
  changeCreateActionState,
  addNewFileTab,
  removeTab,
  onTabChange,
  contractNameChange,
  onChangeFileName,
  onCompileFileLocally
} from '../../components/CodeEditor/codeEditor.actions';
import { extAbi, error } from './codeEditorMock';

describe('CodeEditor: action', () => {

  test('change create action state', () => {
    expect(changeCreateActionState(false, 'abc', 0)).toMatchSnapshot();
  });

  describe('compile code', () => {

    test('request', () => {
      expect(compileCodeFromEditor('code')).toMatchSnapshot();
    });

    test('success', () => {
      expect(compileCodeFromEditorSuccess(extAbi)).toMatchSnapshot();
    });

    test('failure', () => {
      expect(compileCodeFromEditorFailure(error)).toMatchSnapshot();
    });

  })

  test('add new tab', () => {
    expect(addNewFileTab('New file', 'content')).toMatchSnapshot();
  });

  test('remove tab', () => {
    expect(removeTab(0)).toMatchSnapshot();
  });

  test('change tab', () => {
    expect(onTabChange(0, 1)).toMatchSnapshot();
  });

  test('change contract name', () => {
    expect(contractNameChange('Abc')).toMatchSnapshot();
  });

  test('on file name change', () => {
    expect(onChangeFileName('New modified file name')).toMatchSnapshot();
  });

  test('compile file locally', () => {
    expect(onCompileFileLocally(error)).toMatchSnapshot();
  });

});