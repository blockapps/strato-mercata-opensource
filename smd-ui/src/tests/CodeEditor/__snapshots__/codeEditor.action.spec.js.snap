// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeEditor: action add new tab 1`] = `
Object {
  "fileContent": "content",
  "fileName": "New file",
  "type": "ADD_NEW_TAB",
}
`;

exports[`CodeEditor: action change contract name 1`] = `
Object {
  "contractName": "Abc",
  "type": "EDITOR_CONTRACT_NAME_CHANGE",
}
`;

exports[`CodeEditor: action change create action state 1`] = `
Object {
  "createActionEnable": false,
  "index": 0,
  "sourceCode": "abc",
  "type": "CODE_EDITOR_CHANGE_CREATEACTION",
}
`;

exports[`CodeEditor: action change tab 1`] = `
Object {
  "nextTab": 1,
  "prevTab": 0,
  "type": "ON_TAB_CHANGE",
}
`;

exports[`CodeEditor: action compile code failure 1`] = `
Object {
  "error": "src:6:77: Error: Expected token Semicolon got 'RBrace'
{ Sheep sheep = new Sheep(name, sheepDNA); clones.push(sheep); return sheep }
                                                                            ^
",
  "type": "CODE_EDITOR_COMPILE_FAILURE",
}
`;

exports[`CodeEditor: action compile code request 1`] = `
Object {
  "code": "code",
  "codeType": undefined,
  "type": "CODE_EDITOR_COMPILE_REQUEST",
}
`;

exports[`CodeEditor: action compile code success 1`] = `
Object {
  "response": Object {
    "src": Object {
      "Cloner": Object {
        "funcs": Object {
          "cloneSheep": Object {
            "args": Object {
              "name": Object {
                "dynamic": true,
                "index": 0,
                "type": "String",
              },
            },
            "selector": "0072c6a9",
            "vals": Object {
              "#0": Object {
                "index": 0,
                "type": "Address",
              },
            },
          },
        },
        "vars": Object {
          "clones": Object {
            "atBytes": 0,
            "dynamic": true,
            "entry": Object {
              "typedef": "Sheep",
            },
            "public": true,
            "type": "Array",
          },
          "sheepDNA": Object {
            "atBytes": 32,
            "dynamic": true,
            "public": true,
            "type": "String",
          },
        },
      },
      "Sheep": Object {
        "constr": Object {
          "_dna": Object {
            "dynamic": true,
            "index": 1,
            "type": "String",
          },
          "_name": Object {
            "dynamic": true,
            "index": 0,
            "type": "String",
          },
        },
        "funcs": Object {
          "geneticallyModify": Object {
            "args": Object {
              "_dna": Object {
                "dynamic": true,
                "index": 0,
                "type": "String",
              },
            },
            "selector": "0e3668f8",
            "vals": Object {},
          },
        },
        "vars": Object {
          "dna": Object {
            "atBytes": 32,
            "dynamic": true,
            "public": true,
            "type": "String",
          },
          "name": Object {
            "atBytes": 0,
            "dynamic": true,
            "public": true,
            "type": "String",
          },
        },
      },
    },
  },
  "type": "CODE_EDITOR_COMPILE_SUCCESS",
}
`;

exports[`CodeEditor: action compile file locally 1`] = `
Object {
  "compileError": "src:6:77: Error: Expected token Semicolon got 'RBrace'
{ Sheep sheep = new Sheep(name, sheepDNA); clones.push(sheep); return sheep }
                                                                            ^
",
  "type": "ON_COMPILE_FILE_LOCALLY",
}
`;

exports[`CodeEditor: action on file name change 1`] = `
Object {
  "name": "New modified file name",
  "type": "CHANGE_FILE_NAME",
}
`;

exports[`CodeEditor: action remove tab 1`] = `
Object {
  "index": 0,
  "type": "REMOVE_TAB",
}
`;
